CMakeLists.txt

```diff
@@ -512,35 +535,48 @@ if(runtime_cxxmodules)
     set(modules_idx_cmd COMMAND ${ld_library_path}=${library_output_dir}:$ENV{${ld_library_path}}
                                 ROOTIGNOREPREFIX=1 ROOT_HIST=0 $<TARGET_FILE:root.exe> -l -q -b)
   endif()
+  if(ROOT_ENABLE_PROJECTS STREQUAL "All")   
   add_custom_command(OUTPUT ${library_output_dir}/modules.idx
                      COMMAND ${CMAKE_COMMAND} -E remove -f modules.idx modules.timestamp
                      ${modules_idx_cmd}
                      WORKING_DIRECTORY ${library_output_dir}
-                     DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts
+                       DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts
```

Comment:
We should have a function which can filter out non-existing dependencies. Like `root_filter_enabled_dependencies` that takes a list of dependencies and remove the ones that the user did not enabled.

Reply:
fixed